### DISCUSSION TITLE
Retrieve URL raw data (source(

### DISCUSSION THREAD

**Discussion by joejackson3** [KevinnZou/compose-webview-multiplatform#107] [Q&A]
Is it possible to retrieve the plain text of the target URL with compose-webview-multiplatform?
I have a very simple text file saved at a particular URL that I want to retrieve in my KMP project.
I'm seriously struggling to achieve this with KTOR as it has an issue with certain SSL certificates, which I don't fully understand.
I can easily load my URL with Webview as usual and it displays fine, so webview seems to handle that particular SSL certificate without any problems.

**Comment by KevinnZou** [comment]
@joejackson3 If I understand correctly, you can easily realize your needs by using the webview. You just need to load the URL and use `evaluateJavascript` to retrieve the web content by JS after the loading is finished.

**Reply by joejackson3** [reply]
Thanks. My problem is even simpler than I was able to explain above.
This is about a simple text file (.txt) with just a single line of text. It does not containt  HTML or JS.
Is there a function within your library to just read the plain text of the URL object I'm retrieving?

**Reply by KevinnZou** [reply]
If your text file is on the server and can only be retrieved by URL, then you have to use the `evaluateJavascript` to get the content. But it will be quite simple, you can refer to [this](https://github.com/KevinnZou/compose-webview-multiplatform/issues/86#issuecomment-1894830731) for example.
